### DISCUSSION TITLE
Can the console output of a task be filtered?

### DISCUSSION THREAD

**Discussion by smarter** [sbt/sbt#6252] [Q&A]
https://www.scala-sbt.org/1.x/docs/Howto-Logging.html mentions that the `extraLoggers` key can be used to customize logging, but what if I want to prevent the default logger from outputting something logged by a task, is there a way to override it? This could be useful to reduce the amount of noise in the console output if a task is too verbose.

**Comment by eed3si9n** [comment]
> This could be useful to reduce the amount of noise in the console output if a task is too verbose.

Are you planning to do this just for your own build, or from a plugin?
It may not be a great situation if plugin implementations started suppressing each others outputs.

I'm guessing that you could provide your own [Streams](https://github.com/sbt/sbt/blob/v1.4.6/tasks-standard/src/main/scala/sbt/std/Streams.scala#L85-L96) implementation that creates a custom logger for task evaluations, and set it on `Global / streamsManager`.

Ethan removed log4j last year in https://github.com/sbt/sbt/pull/5731, so that might also be a useful reference on what classes are involved in logging.

**Reply by smarter** [reply]
> Are you planning to do this just for your own build, or from a plugin?

This isn't something I would put in a plugin no.

>  that might also be a useful reference on what classes are involved in logging.

Yeah there's a ton of them and I'm completely overwhelmed by the complexity of it all.
